### ISSUE TITLE
CATS does not save the model

### ISSUE THREAD

**Issue by SergeevVladislav** [VowpalWabbit/vowpal_wabbit#3116]
[003-Cats-Tutorial.ipynb.zip](https://github.com/VowpalWabbit/vowpal_wabbit/files/6748242/003-Cats-Tutorial.ipynb.zip)
I train a multi-armed bandit model for a continuous action space.
After training, I save the model, but after unloading the model, the progress is not saved or the model does not predict on old examples.

`
vw.predict(ex)
`
Result:
`
(13.885909080505371, 0.40625)
`

```
vw.save("mab.model")

model = pyvw.vw("mab.model")

model.predict(ex)
```
Result:
`
0.0
`

If I use loading with `--quiet -i`, then a bug comes when predicting

```
RuntimeError: Continuous actions labels require the first word to be ca

```

#### Environment
VW 8.10.2 on MAC OS with the Python bindings

**Comment by kdubovikov** [comment]
We have noticed that it works ok from command line, but Python wrapper is not working properly

**Comment by jackgerrits** [comment]
Just a quick correction - this doesn't load a model - it will try and read a single data file "mab.model" and fail since it doesn't exist.
```Python
model = pyvw.vw("mab.model")
```

You had it right the second time:
```Python
model = pyvw.vw("-i mab.model")
```

However, the error message you got implies it failed to parse the given example. The string passed to predict is:
```json
{"c": {"room=Bedroom": 1, "time_of_day=afternoon": 1}}
```

The reason this fails is because when you save a VW model it only saves things related to the model produced and not how it parses input. This is extra confusing when using the Python interface.

When you created that `vw` instance in the notebook the code to produce the command line was:
```Python
vw = pyvw.vw("--cats " + str(num_actions) + "  --bandwidth " + str(bandwidth) + " --min_value 0 --max_value 32 --json --chain_hash --coin --epsilon 0.2 -q ::")
```

You'll notice the `--json` in there. 

If you update your loading command to:
```Python
model = pyvw.vw("-i mab.model --json")
```

Then things should work as intended.

To actually discover for yourself that `--json` wasn't in there you can inspect the contents of the model file. This is not well supported directly in Python but we can invoke VW as if it were CLI by the following:
```Python
model = pyvw.vw("-i mab.model --readable_model readable.txt")
del model
```

We can then see in `readable.txt` the contents of the command line which got saved on the line starting with `options:`
```
options: --bandwidth 1 --binary --cats 32 --cats_pdf 32 --cats_tree 32 --cb_explore_pdf --coin --epsilon 0.200000002980232 --get_pmf --max_value 32 --min_value 0 --pmf_to_pdf 32 --quadratic :: --sample_pdf --tree_bandwidth 1
```

**Comment by kdubovikov** [comment]
@jackgerrits we have tried `model = pyvw.vw("-i mab.model --json")` and models give out predictions correctly, **but** it still fails to save the model correctly. After we load the model, `vw.get_sum_loss()` returns 0, which we suppose is not the expected behaviour when you try to continue learning using an existing model

@SergeevVladislav can provide an example for that if needed

**Comment by jackgerrits** [comment]
Currently the model does not save most of the progressive validation metrics which `sum_loss` is one of them. So it essentially is just the sum loss for this "session", and does not carry over when a model is loaded. This should have no effect on the predictions or training though.

However, it's important that if you wish to continue training a model after loading it you need to save it with the `--save_resume` flag. This saves additional state required to resume the training process.

**Comment by SergeevVladislav** [comment]
Thank you for answering!
But when I initialize the model with the flag `--save_resume` in
```
vw = pyvw.vw("--cats " + str(num_actions) + "  --bandwidth " + str(bandwidth) + " --min_value 0 --max_value 32 --json --chain_hash --coin --epsilon 0.2 -q :: --save_resume")
```
and after loading model `model = pyvw.vw("-i aws.model --json --save_resume")` I get different predictions:

```
model.predict(ex)
```

Result: `(0.0012017497792840004, 0.0062500000931322575)`, but expected `(19.793601989746094, 0.40625)`

**Comment by jackgerrits** [comment]
Interesting, okay there may be a bug in save/load for continuous actions. @bassmang is going to take a look

**Comment by bassmang** [comment]
@SergeevVladislav  this issue is now be resolved with #3150
